### PR TITLE
Backport cluster address usage in DCS

### PR DESCRIPTION
**Pull request by KrishnaIyer**
#### Summary

Backport https://github.com/TheThingsIndustries/lorawan-stack/pull/3457

#### Changes

See https://github.com/TheThingsIndustries/lorawan-stack/pull/3457

#### Testing

See https://github.com/TheThingsIndustries/lorawan-stack/pull/3457/files

##### Regressions

See https://github.com/TheThingsIndustries/lorawan-stack/pull/3457/files

#### Checklist

- [ ] Scope: The referenced issue is addressed, there are no unrelated changes.
- [ ] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/deviceclaimingserver/enddevices/config.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ const JSClientConfigurationName = "config.yml"
 
 // NetworkServer contains information related to the Network Server.
 type NetworkServer struct {
-	Hostname string      `name:"hostname" description:"Hostname of the Network Server. Must not contain a port"`
+	Hostname string      `name:"hostname" description:"Hostname of the Network Server. Must not contain a port. 'cluster-id' and 'cluster-id-address-template' values override this field in enterprise deployments"` //nolint:lll
```
Comment (johanstokking) [inline pkg/deviceclaimingserver/enddevices/config.go]: This is not relevant for TTSOS right?

Comment (johanstokking) [reply pkg/deviceclaimingserver/enddevices/config.go]: On another note; these config keys are not declared here but in `cluster`

**Review by nicholaspcr — APPROVED** [review]

File: pkg/deviceclaimingserver/enddevices/config.go [hunk 2]
```diff
@@ -28,7 +28,7 @@ const JSClientConfigurationName = "config.yml"
 
 // NetworkServer contains information related to the Network Server.
 type NetworkServer struct {
-	Hostname string      `name:"hostname" description:"Hostname of the Network Server. Must not contain a port"`
+	Hostname string      `name:"hostname" description:"Hostname of the Network Server. Must not contain a port."` //nolint:lll
```
Comment (johanstokking) [inline pkg/deviceclaimingserver/enddevices/config.go:31]:
```suggestion
	Hostname string      `name:"hostname" description:"Hostname of the Network Server. Must not contain a port"` //nolint:lll
```

**Review by johanstokking — APPROVED** [review]

**Comment by KrishnaIyer** [comment]
This is no longer relevant due to https://github.com/TheThingsIndustries/lorawan-stack/pull/3467

**Pull request closed without merging**